cobc/parser.y

```diff
@@ -8112,6 +8142,17 @@ usage:
   }
 ;
 
+_to_program_type:
+  /* empty */		{ $$ = NULL; }
+| _to PROGRAM_NAME	{ $$ = $2;   }
+;
+
+_to_type_name:
+  /* empty */ %prec SHIFT_PREFER { $$ = NULL; }
+| TYPEDEF_NAME                   { $$ = $1;   }
+| TO TYPEDEF_NAME                { $$ = $2;   }
```

Comment:
We can keep that but should get back to the original, simpler version:

```yacc
_to_type_name:
  /* empty */	{ $$ = NULL; }
| _to type_name { $$ = $2;   }
;
```

To do so, we should likely adjust the scanner to return TYPE_NAME - similar like it is done for the PROGRAM_NAME; that should be done in GC3.x and merged, but please still add a TODO entry for it.